### DISCUSSION TITLE
Cannot load local URDF file via File field

### DISCUSSION THREAD

**Discussion by christian-nils** [foxglove/community#857] [Q&A] [answered]
Hello,

A bit similar to the issue https://github.com/foxglove/studio/issues/5382, I cannot load local URDF files. Except that the proposed resolution (i.e., select URL and then use `file://` does not work for me). For both the desktop app and the web app I experience the issue. On the desktop app I saw in the Dev Console that the app was `Not allowed to load local resource`. So I use https://www.npmjs.com/package/http-server instead now and it works using the URL setting.

Does anyone know how to use the File setting instead? It would be better than having to start up a local http server to serve the static files.

Desktop app version: 1.74.1
Web app version: 1.74.1+fd1796d.767f69b on Google Chrome 118.0.5993.88
OS: Fedora Linux 38 (Workstation Edition)

Thanks in advance!

Christian-Nils

**Comment by achim-k** [comment]
What file path do you use? The following works for me:
![image](https://github.com/foxglove/community/assets/9250155/7ff16d73-8595-4de6-8d19-f81c6597a382)

> On the desktop app I saw in the Dev Console that the app was `Not allowed to load local resource`

This sounds weird. The Desktop app should be allowed to load local resources :thinking:

**Reply by christian-nils** [reply]
Hi @achim-k,

Thanks for your quick assistance. First, I need to apologize, I have indeed wrote "Desktop app", I meant "Web app".

So now, focusing on the Desktop app (installed via snap), I can load urdf models that are placed on my home directory. However, when I try to load a file that is stored on another disk I get the error, see below:


https://github.com/foxglove/community/assets/2658014/ec4a77fd-7c26-4025-a420-a9b2466935b6


Any ideas on how to debug this? :-) Thanks for the help again.

//CN

**Reply by christian-nils** [reply]
Hi again,

I wonder if this is because of the issue mentioned in https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1643706. If this is the case, then maybe Foxglove team cannot fix the issue. Could it be worth adding a note about it in the documentation for people using the snap version?

Otherwise, could you consider publishing Foxglove Studio as a flatpak or appimage package?

Thanks again

**Reply by achim-k** [reply, marked as answer]
> I wonder if this is because of the issue mentioned in https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1643706. If this is the case, then maybe Foxglove team cannot fix the issue. Could it be worth adding a note about it in the documentation for people using the snap version?

Oh wow, I was not aware of this issue but I just tested it and I cannot even access a URDF that is under `/tmp` :sob: This is really a severe limitation of the Studio snap.

> Otherwise, could you consider publishing Foxglove Studio as a flatpak or appimage package?

Good news, we are actively working on publishing Studio as Flatpak. But that will probably still take a few weeks, if you can't wait that long, contact me on our public slack and I will send you the `1.73.0` release packed as Flatpak (we used that for running Studio on the SteamDeck at ROSCon last week)

**Reply by christian-nils** [reply]
Thanks! This is not urgent, I'll place my files in the home directory till you release a flatpak! Thanks again for the quick help!

**Reply by defunctzombie** [reply]
Until we get around to publishing a flatpak we do offer a debian package which does not have these limitations.

**Reply by christian-nils** [reply]
That is right! But in my case, I cannot use that package as I am on Fedora.
But it does not matter much for now since we know it does work using the home directory for the time being.
